pyvista/core/utilities/points.py

```diff
@@ -459,3 +464,90 @@ def vector_poly_data(orig, vec):
     pdata.GetPointData().SetActiveScalars(name)
 
     return pyvista.PolyData(pdata)
+
+
+def principal_axes(points: MatrixLike[float], *, return_sizes: bool = False):
+    """Compute the principal axes of a set of points.
+
+    Principal axes are orthonormal vectors that best fit a set of points. The axes
+    are also known as the principal components in Principal Component Analysis (PCA),
+    or the right singular vectors from the Singular Value Decomposition (SVD).
+
+    The axes are computed as the eigenvectors of the covariance matrix from the
+    mean-centered points, and are processed to ensure that they form a right-handed
+    coordinate frame.
+
+    The axes explain the total variance of the points. The first axis explains the
+    largest percentage of variance, followed by the second axis, followed again by
+    the third axis which explains the smallest percentage of variance. The axes may
+    be used to build an oriented bounding box or to align the points to another set
+    of axes (e.g. the world XYZ axes).
+
+    .. note::
+        The computed axes are not unique, and the sign of each axis direction can be
+        arbitrarily changed.
+
+    .. versionadded:: 0.45.0
+
+    Parameters
+    ----------
+    points : MatrixLike[float]
+        Nx3 array of points.
+
+    return_sizes : bool, default: False
+        If ``True``, also returns the axes sizes. The sizes are computed as the square
+        root of the eigen vectors of the mean-centered covariance matrix.
+
+    Returns
+    -------
+    numpy.ndarray
+        3x3 orthonormal array with the principal axes as row vectors.
+
+    numpy.ndarray
+        Three-item array of the axes sizes.
+
+    Examples
+    --------
+    Create a mesh with points that have the largest variation in ``X``,
+    followed by ``Y``, then ``Z``.
+
+    >>> import pyvista as pv
+    >>> mesh = pv.ParametricEllipsoid(xradius=10, yradius=5, zradius=1)
+    >>> p = pv.Plotter()
+    >>> _ = p.add_mesh(mesh)
+    >>> _ = p.show_grid()
+    >>> p.show()
+
+    Compute its principal axes.
+
+    >>> principal_axes = pv.principal_axes(mesh.points)
+    >>> principal_axes
+    array([[-1.0000000e+00,  1.6606515e-10,  1.9055901e-10],
+           [-1.6606515e-10, -1.0000000e+00, -5.0257620e-10],
+           [ 1.9055901e-10, -5.0257620e-10,  1.0000000e+00]], dtype=float32)
+
+    Note that the principal axes have ones along the diagonal and zeros
+    in the off diagonals. This indicates that the first principal axis is
```

Comment:
I'm less sure about this one but I usually see this spelled either `offdiagonal` or `off-diagonal`, never separately.

Reply:
```suggestion
    in the off-diagonal. This indicates that the first principal axis is
```